src/main/scala/ir/Program.scala

```diff
@@ -116,52 +120,265 @@ class Program(var procedures: ArrayBuffer[Procedure], var mainProcedure: Procedu
     initialMemory = initialMemoryNew
   }
 
+  class ILUnorderedIterator(private val begin: Program) extends Iterator[CFGPosition] {
```

Comment:
@l-kent Do you want to keep this in? it's just a convenient way to .map() and .filter() etc. the program

Reply:
I think you should make it clear what order it traverses the blocks within a procedure in, for instance (otherwise it could be prone to misuse). Would the idea to just be use .map() with the results of an analysis? I'm not entirely sure how useful that is

Reply:
I just commented it, its a very special case visitor I guess, probably only useful in very specific cases to map over the whole program, e.g. collect indirect calls and see if they were resolved. E.g. a simpler way to get the analysis domain.